src/Components/OATPropertyEditor/Internal/PropertyList/Internal/PropertyListItem/Internal/PropertyListItemChildHost/PropertyListItemChildHost.tsx

```diff
@@ -129,6 +129,20 @@ const PropertyListItemChildHost: React.FC<IPropertyListItemChildHostProps> = (
                     indexKey={`${indexKey}.0`}
                     item={propertyItem.schema}
                     level={level}
+                    onUpdateKey={(key) => {
+                        // update the schema for the item
+                        const schemaCopy = deepCopy(propertyItem.schema);
+                        schemaCopy.mapKey = key;
+                        // send updated schema to parent
+                        onUpdateSchema(schemaCopy);
+                    }}
+                    onUpdateValue={(value) => {
```

Comment:
This function is also used in `PropertyListItemMapChild.tsx` it could even be part of a utils file

Reply:
in this case I think it is fine to be in both spots. It follows the same pattern as the rest of these callbacks for the other child properties so I'd rather not obfuscate it by moving to a utility.